components/clp-mcp-server/tests/test_clp_connector.py

```
@@ -110,27 +110,45 @@ async def test_wait_query_completion_failure_cases(
+    ]
+    mock_collection = AsyncMock()
+    mock_collection.find = MagicMock(return_value=_aiter(mock_docs))
+
+    with patch.object(connector, "_results_cache", {"12": mock_collection}):
+        results = await connector.read_results("12")
+
+    assert len(results) == len(mock_docs)
+
+
+@pytest.mark.asyncio
+async def test_read_results_supports_legacy_docs(mock_clp_config: Any) -> None:
```

Ruff (0.16.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `mock_clp_config`

(ANN401)